airbyte-webapp/src/packages/cloud/services/freeConnectorProgram.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from "react-query";
+
+import { useCurrentWorkspace } from "hooks/services/useWorkspace";
+import { useDefaultRequestMiddlewares } from "services/useDefaultRequestMiddlewares";
+
+import { webBackendGetFreeConnectorProgramInfoForWorkspace } from "../lib/domain/freeConnectorProgram/api";
+import { useConfig } from "./config";
+
+export const useFreeConnectorProgramInfo = () => {
+  const { workspaceId } = useCurrentWorkspace();
```

Comment:
Usually I use `useCurrentWorkspaceId()` to just get the workspace ID. It's synchronous and just returns the id from the URL. `useCurrentWorkspace()` calls a suspense query, so it could create a waterfall request dependency in the worst case.